packages/ipfs-daemon/src/make-storage-backends.ts

```diff
@@ -0,0 +1,40 @@
+import DatastoreLevel from 'datastore-level';
```

Comment:
I think the use of the word `local` is too ambiguous the way you have it here.

It seems like the most straightforward thing is an env var for each data store with enumerated options. Something like...

```sh
IPFS_ROOT_STORE=default
IPFS_BLOCKS_STORE=s3
IPFS_KEYS_STORE=default
IPFS_PINS_STORE=default
IPFS_DATASTORE_STORE=default
```

If one of these is omitted, it becomes default. If one of these is s3, the s3 related env vars are required or we throw. In this case it is probably better to remove `IPFS_S3_REPO_ENABLED` as well.

Maybe we name it `IPFS_<datastore>_BACKEND` instead?

Reply:
@valmack `IPFS_BACKEND_<datastore>` sounds fine?

Now configuration is segregated. Local prefix is fully optional. I find it useful still, as a first-tier directory can be unavailable.